src/main/scala/nl.knaw.dans.easy.solr4files/components/Solr.scala

```diff
@@ -54,22 +59,61 @@ trait Solr extends DebugEnhancedLogging {
     }
   }
 
-  def deleteDocuments(query: String): Try[Unit] = {
-    Try(solrClient.deleteByQuery(new SolrQuery {set("q", query) }.getQuery))
-      .flatMap(checkUpdateResponseStatus)
+  def deleteDocuments(query: String): Try[UpdateResponse] = {
+    val q = new SolrQuery {
+      set("q", query)
+    }
+    Try(solrClient.deleteByQuery(q.getQuery))
+      .flatMap(checkResponseStatus)
       .recoverWith {
-        case t: HttpSolrClient.RemoteSolrException if t.getRootThrowable.endsWith("ParseException") =>
+        case t: HttpSolrClient.RemoteSolrException if isPareException(t) =>
           Failure(SolrBadRequestException(t.getMessage, t))
-        case t: SolrServerException =>
-          Failure(SolrDeleteException(query, t))
         case t =>
           Failure(SolrDeleteException(query, t))
       }
   }
 
-  def commit(): Try[Unit] = {
+  private def toJson(solrDocumentList: SolrDocumentList, query: SolrQuery): Try[String] = {
+    def toScalaMap(fieldValueMap: util.Map[String, AnyRef]) = {
+      // asScala is not good enough, serialization needs scala case classes
+      // less verbose than TypeHints, after all we need only one direction
+      fieldValueMap.keySet().asScala.map(key => key -> fieldValueMap.get(key))
+    }
+
+    val found = solrDocumentList.getNumFound
+    Try(write(Map("header" -> Map(
```

Comment:
is this whole thing supposed to represent JSON? If so, have you tried using the library's syntax instead. See also https://github.com/DANS-KNAW/easy-stage-dataset/blob/master/src/main/scala/nl.knaw.dans.easy.stage/lib/JSON.scala